core/eolearn/core/eodata.py

```diff
@@ -731,3 +734,92 @@ def plot(
             config=config,
             **kwargs,
         )
+
+
+@dataclass
+class PatchGeneratorConfig:
+    """Dataclass containing a more complex setup of the PatchGenerator class."""
+
+    raster_shape: Tuple[int, int] = (300, 400)
+    bbox: BBox = field(init=False, repr=False)
+
+    timestamps_periods: int = 15
+    timestamps_range: Tuple[str, str] = ("2019-01-01", "2019-12-31")
+    timestamps: List[dt.datetime] = field(init=False, repr=False)
+
+    num_random_features: int = 5
+    allowed_feature_types: List[FeatureType] = field(default_factory=list)
+    max_integer_value: int = 256
+
+    def __post_init__(self):
+        self.bbox = BBox((0, 0, *self.raster_shape), crs=CRS.UTM_33N)
+        self.timestamps = pd.date_range(*self.timestamps_range, periods=self.timestamps_periods)
+        if not self.allowed_feature_types:
+            self.allowed_feature_types = [ftype for ftype in FeatureType if ftype.is_raster()]
+
+
+class PatchGenerator:
+    """A class for generating EOPatches with dummy data."""
```

Comment:
Might try to convert this back to a function now... theres a lot of definitions/docstrings ballast which might reduce the code footprint.

Reply:
Agreed. One should (in general) use classes in 3 cases:
1. It has a mutable state that the methods interact with
2. Inheritance simplifies things (happens less often than OOP people believe)
3. (python specific) you want to organize things under a common namespace

Try switching to functions and I'll do a more thorough review then.

Reply:
converted, much nicer now